packages/babel-parser/src/parser/util.js

```diff
@@ -211,7 +213,7 @@ export default class UtilParser extends Tokenizer {
     oldState: State = this.state.clone(),
   ):
     | TryParse<T, null, false, false, null>
-    | TryParse<T | null, SyntaxError, boolean, false, State>
```

Comment:
I think the parser throws a ParsingError: https://github.com/babel/babel/blob/6c8b2336f6216479215114585c3e0777f8113a6c/packages/babel-parser/src/parser/error.js#L20